phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -1148,115 +1235,203 @@ private PTable getTable(RegionScanner scanner, long clientTimeStamp, long tableT
                 Bytes.contains(tagPhoenixTTL, VIEW_MODIFIED_PROPERTY_BYTES);
+        builder.setViewModifiedPhoenixTTL(oldTable != null ?
+            oldTable.hasViewModifiedPhoenixTTL() || viewModifiedPhoenixTTL : viewModifiedPhoenixTTL);
 
         Cell lastDDLTimestampKv = tableKeyValues[LAST_DDL_TIMESTAMP_INDEX];
         Long lastDDLTimestamp = lastDDLTimestampKv == null ?
            null : PLong.INSTANCE.getCodec().decodeLong(lastDDLTimestampKv.getValueArray(),
                 lastDDLTimestampKv.getValueOffset(), SortOrder.getDefault());
+        builder.setLastDDLTimestamp(lastDDLTimestampKv != null ? lastDDLTimestamp :
+            oldTable != null ? oldTable.getLastDDLTimestamp() : null);
 
         Cell changeDetectionEnabledKv = tableKeyValues[CHANGE_DETECTION_ENABLED_INDEX];
         boolean isChangeDetectionEnabled = changeDetectionEnabledKv != null
             && Boolean.TRUE.equals(PBoolean.INSTANCE.toObject(changeDetectionEnabledKv.getValueArray(),
             changeDetectionEnabledKv.getValueOffset(),
             changeDetectionEnabledKv.getValueLength()));
+        builder.setIsChangeDetectionEnabled(changeDetectionEnabledKv != null ?
+            isChangeDetectionEnabled : oldTable != null && oldTable.isChangeDetectionEnabled());
 
         Cell schemaVersionKv = tableKeyValues[SCHEMA_VERSION_INDEX];
         String schemaVersion = schemaVersionKv != null ? (String) PVarchar.INSTANCE.toObject(
                 schemaVersionKv.getValueArray(), schemaVersionKv.getValueOffset(), schemaVersionKv.getValueLength())
                 : null;
+        builder.setSchemaVersion(schemaVersion != null ?
+            schemaVersion : oldTable != null ? oldTable.getSchemaVersion() : null);
+
+        Cell externalSchemaIdKv = tableKeyValues[EXTERNAL_SCHEMA_ID_INDEX];
+        String externalSchemaId = externalSchemaIdKv != null ?
+            (String) PVarchar.INSTANCE.toObject(externalSchemaIdKv.getValueArray(),
+                externalSchemaIdKv.getValueOffset(), externalSchemaIdKv.getValueLength())
+            : null;
+        builder.setExternalSchemaId(externalSchemaId != null ? externalSchemaId :
+            oldTable != null ? oldTable.getExternalSchemaId() : null);
 
         // Check the cell tag to see whether the view has modified this property
         final byte[] tagUseStatsForParallelization = (useStatsForParallelizationKv == null) ?
                 HConstants.EMPTY_BYTE_ARRAY : CellUtil.getTagArray(useStatsForParallelizationKv);
         boolean viewModifiedUseStatsForParallelization = (PTableType.VIEW.equals(tableType)) &&
                 Bytes.contains(tagUseStatsForParallelization, VIEW_MODIFIED_PROPERTY_BYTES);
+        builder.setViewModifiedUseStatsForParallelization(viewModifiedUseStatsForParallelization ||
+            (oldTable != null && oldTable.hasViewModifiedUseStatsForParallelization()));
 
+        boolean setPhysicalName = false;
```

Comment:
Could you explain more about how that would be a problem? Note that oldTable is only not-null when getTableFromCells is called during schema registry export during an ALTER when we're trying to create a PTable that merges in the Cells from the ALTER statement + the existing PTable. This is just trying to say that if we didn't already figure out the physical name from any PHYSICAL linking rows passed in as Cells, grab it from the previous (pre-ALTER) version of the PTable we got from the client. 

In the schema-registry export during CREATE case, oldTable is null and we should populate everything from the Cells we're passed in. 

In all other cases (we call this during a getTable() cache refresh) it should work the same as it's always worked. 

That said, I found the physicalNames/physicalTable/parentName APIs a bit confusing so if I've missed something please let me know. (And of course this will probably have to be tweaked some when your transform work gets merged in later.)